feelpp/feel/feelcrb/pbdw.hpp

```diff
@@ -0,0 +1,355 @@
+class PBDW : CRBDB
+{
+public:
+    using super_type = CRBDB;
+    using reducedspace_type = RBSpace;
+    using reducedspace_ptrtype = std::shared_ptr<reducedspace_type>;
+    using space_type = typename reducedspace_type::fespace_type;
+    using space_ptrtype = std::shared_ptr<space_type>;
+    using mesh_type = typename space_type::mesh_type;
+    using node_t = typename mesh_type::node_type;
+    using element_type = typename space_type::element_type;
+    using vectorN_type = Eigen::VectorXd;
+    using matrixN_type = Eigen::MatrixXd;
+    using sensorbase_type = SensorBase<space_type>;
+    using sensorbase_ptrtype = std::shared_ptr<sensorbase_type>;
+    static const int nDim = space_type::mesh_type::nDim;
+
+    /**
+     * Constructor for the online phase
+     * @param name Name of pbdw
+     * @param l Loading type (rb,fe,all)
+     * @param uuid Uuid to use for the db
+     */
+    explicit PBDW(std::string const& name,
+                  crb::load l = crb::load::rb,
+                  uuids::uuid const& uuid = uuids::nil_uuid());
+    /**
+     * Constructor for the offline phase
+     * @param name Name of pbdw
+     * @param XR Reduced Basis
+     * @param sigmas Sensors to use
+     * @param uuid Uuid to use for the db
+     */
+    PBDW(std::string const& name,
+         reducedspace_ptrtype const& XR,
+         std::vector<sensorbase_ptrtype> const& sigmas,
+         uuids::uuid const& uuid = uuids::nil_uuid());
+    int dimensionN() const { return M_N; } /**< Dimension of Reduced Basis */
+    int dimensionM() const { return M_M; } /**< Number of sensors */
+    int dimension() const { return M_N+M_M; } /**< Dimension of PBDW */
+    space_ptrtype functionSpace() const { return M_XR->functionSpace(); } /**< Function Space */
+    matrixN_type matrix() const { return M_matrix; } /**< Matrix of PBDW */
+    void offline(); /** Do offline phase */
+    /**
+     * Do online phase
+     * @param yobs Observation of sensors
+     * @return Coefficients of solution
+     */
+    vectorN_type online(vectorN_type const& yobs) const; /**< Do online phase */
+    /**
+     * Retrieve solution
+     * @param yobs Observation of sensors
+     * @return Solution
+     */
+    element_type solution(vectorN_type const& yobs) const; /**< Retrieve solution */
+
+protected:
+    void loadDB( std::string const& filename, crb::load l ) override;
+    void saveDB() override;
+
+
+private:
+    std::string M_name;
+    reducedspace_ptrtype M_XR;
+    std::vector<sensorbase_ptrtype> M_sigmas;
+    matrixN_type M_matrix;
+    int M_M;
+    int M_N;
+
+    bool M_rebuildDb;
+    int M_dbLoad;
+    std::string M_dbFilename;
+    std::string M_dbId;
+    crb::stage M_stage;
+};
+
+template<typename RBSpace>
+PBDW<RBSpace>::PBDW(std::string const& name,
+                    crb::load l,
+                    uuids::uuid const& uuid):
+    super_type(name, "pbdw", uuid),
+    M_name(name),
+    M_rebuildDb(boption("pbdw.rebuild-database")),
```

Comment:
would it be possible to change these parameters from the interface and not from the command line ?
perhaps add a parameter which is the variable_map for the command line options to specify this construction.
otherwise use default sensible values

Reply:
idem for all options `pbdw.`

Reply:
Ok I have added parameters for them. The only remaining are the tolerance and the maximum number of basis in GEIM, but those are not used in the constructor and have a setter.